lib/internal/streams/buffer_list.js

```diff
@@ -137,7 +137,10 @@ module.exports = class BufferList {
     while (p = p.next) {
       const buf = p.data;
       const nb = (n > buf.length ? buf.length : n);
-      buf.copy(ret, ret.length - n, 0, nb);
+      if (nb === buf.length)
+        ret.set(buf, ret.length - n);
+      else
+        ret.set(new Uint8Array(buf.buffer, buf.byteOffset, nb), ret.length - n);
```

Comment:
how does this differ from copy?

Reply:
It's a tad faster.

Reply:
yes, just confused as two how it can be faster. Some V8 thingy?

Reply:
yes. v8 can optimize the stuff it provides, but our buffer copy is basically a black box to it, since its written in c++. we could probably optimize our copy impl to use typedarray's set if there are no source offsets passed.

Reply:
wouldn't it better to apply this there then?